src/lua/fio.lua

```diff
@@ -238,51 +239,14 @@ fio.open = function(path, flags, mode)
     return res
 end
 
-fio.pathjoin = function(...)
-    local i, path = 1, nil
-
-    local len = select('#', ...)
-    while i <= len do
-        local sp = select(i, ...)
-        if sp == nil then
-            error("fio.pathjoin(): undefined path part "..i)
-        end
-
-        sp = tostring(sp)
-        if sp ~= '' then
-            path = sp
-            break
-        else
-            i = i + 1
-        end
-    end
-
-    if path == nil then
-        return '.'
-    end
-
-    i = i + 1
-    while i <= len do
-        local sp = select(i, ...)
-        if sp == nil then
-            error("fio.pathjoin(): undefined path part "..i)
-        end
-
-        sp = tostring(sp)
-        if sp ~= '' then
-            path = path .. '/' .. sp
-        end
-
-        i = i + 1
-    end
-
-    path = path:gsub('/+', '/')
-    if path ~= '/' then
-        path = path:gsub('/$', '')
-    end
-
-    return path
+-- Expose several functions implemented as part of the minifio
+-- module.
+for name, func in pairs(minifio.expose_from_fio) do
+    fio[name] = func
 end
```

Comment:
Do you move it out of cycle to test that these specific functions are available?

Reply:
Do you mean the assert right below the cycle?

```lua
assert(type(fio.cwd) == 'function')
assert(type(fio.pathjoin) == 'function')
assert(type(fio.abspath) == 'function')
```

It aims two goals:

* Ensure that the functions are there.
* Make them searchable/greppable, when one is looking over the `fio.lua` file: it is not obvious that they're implemented in another place.

Reply:
I suggest to add the comment with the text of the second bullet to warn future generations (or at least me and @Buristan) of moving these assertions to the loop later.

Reply:
I agree, added a note.

```diff
diff --git a/src/lua/fio.lua b/src/lua/fio.lua
index 6c5c51656..ea66aee22 100644
--- a/src/lua/fio.lua
+++ b/src/lua/fio.lua
@@ -244,6 +244,12 @@ end
 for name, func in pairs(minifio.expose_from_fio) do
     fio[name] = func
 end
+
+-- List of functions exposed from fio, but defined in minifio.
+--
+-- Without the list a reader should be aware of the fio/minifio
+-- spliting in order to find an implementation of a function by
+-- its name. Let's make a reader's life a bit easier.
 assert(type(fio.cwd) == 'function')
 assert(type(fio.pathjoin) == 'function')
 assert(type(fio.abspath) == 'function')
```